wpilibj/src/main/java/edu/wpi/first/wpilibj/can/CANJNI.java

```diff
@@ -29,4 +29,8 @@ public static native void FRCNetCommCANSessionMuxSendMessage(int messageID,
   @SuppressWarnings("MethodName")
   public static native byte[] FRCNetCommCANSessionMuxReceiveMessage(
       IntBuffer messageID, int messageIDMask, ByteBuffer timeStamp);
+
+
+  @SuppressWarnings("MethodName")
+  public static native void GetCANStatus(CANStatus status);
```

Comment:
This is rather non-idiomatic (versus returning a new CANStatus), but I guess it does avoid memory allocations.

Reply:
Yeah I was stuck between which one do to, but I decided I'd rather have the JNI side be non allocating, as we can decide at the WPILib level if we would like the call there to be allocating or not. It just gives us more options. I'm going to do the same thing in a separate PR to the PWM status structure.

Reply:
Would the WPILib level use the same CANStatus object?  I think the fields of what we return from the WPILib level should be final.

Reply:
Yes they would, but we can't make them final without making every call allocating. So I think it's worth it to just leave them. If a user changes them it doesn't affect anything else.